test/servinge2e/service_to_service_test.go

```diff
@@ -89,6 +98,53 @@ func testServiceToService(t *testing.T, ctx *test.Context, namespace string, tc
 		t.Errorf("the Route at domain %s didn't serve the expected text %q: %v", service.Status.URL.URL(), helloworldText, err)
 	}
 
+	// Check if service monitor service is available, at this point it should be present
+	_, err := ctx.Clients.Kube.CoreV1().Services("openshift-serverless").Get(context.Background(), "knative-openshift-metrics", meta.GetOptions{})
+	if err != nil {
+		t.Errorf("error getting service monitor service: %v", err)
```

Comment:
This should be Fatal as the rest of the test won't work (actually true for almost all statements hereafter)

Reply:
Yeah I was thinking about it guess if I move this in a separate test it makes sense.